cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORGenerator.java

```diff
@@ -924,8 +958,25 @@ public void writeBinary(Base64Variant b64variant, byte[] data, int offset,
             return;
         }
         _verifyValueWrite("write Binary value");
+        ByteBuffer bytesRef = null;
+        if (_stringRefs != null) {
```

Comment:
Does StringRefs functionality also apply to binary values? That seems.... odd. Not all binary values represent Strings.

Would it make sense to have separate features for enabling these separately? (I assume it is fine to write combination of references and non-referenced String / binary values).

Reply:
The extension requires computing the index for every binary and text string within a namespace as long as it fits within the minimum size based on the current index value. (i.e. the more bytes required to store the index, the longer the threshold to compute the index for sharing) It doesn't allow you to selectively apply indices with the exception to indefinite length (chunked) strings, which will never have an index computed.

In other words, the only way to exclude binary strings from the index computation is to always write them with the indefinite length encoding.

Reply:
Ok. So it is required by CBOR spec. Interesting choices; esp. not having upper limit. Could be problematic for many binary data cases but I guess that's not my problem :)
(I spec'd/implemented String back refs for Smile format before CBOR was a thing; never even considered shared refs for binary data so this seems.... interesting).

Reply:
I would imagine that if a client has a significant amount of binary data to the point where this becomes an issue they could simply not opt into stringrefs. If clients are concerned about *both* a significant number of shared strings bloating their data *and* large embedded binaries (or large strings for that matter), an option would be to introduce a configuration to have an upper limit on string size before it forces a chunked encoding. (which would bypass the stringref sharing) I would imagine this is very likely very rare, however, and probably not worth the complexity of introducing an option until there's a real-world example that both proves the need for it and provides an example to more effectively create such a feature.